### PR TITLE
Thermal BC for spe11bc

### PR DESCRIPTION
**Pull request by daavid00**
This PR allows to set the no flux and constant temperature on the top (1) and bottom (2) boundaries for the SPE11B/C cases: 
```
BCPROP
1 THERMAL /
2 THERMAL /
/
```

### CODE REVIEW

**Comment by daavid00** [comment]
jenkins build this please

File: opm/models/blackoil/blackoillocalresidualtpfa.hh [hunk 1]
```diff
@@ -602,6 +604,48 @@ public:
...
+            EnergyModule::addHeatFlux(bdyFlux, heatFlux);
+        }
+
+        static_assert(!enableSolvent, "Relevant treatment of boundary conditions must be implemented before enabling.");
+        static_assert(!enablePolymer, "Relevant treatment of boundary conditions must be implemented before enabling.");
+        static_assert(!enableMICP, "Relevant treatment of boundary conditions must be implemented before enabling.");
+
+        // make sure that the right mass conservation quantities are used
+        adaptMassConservationQuantities_(bdyFlux, insideIntQuants.pvtRegionIndex());
+
```
Comment (totto82) [inline opm/models/blackoil/blackoillocalresidualtpfa.hh:648]: Line 642 to 647 can  be removed. This function only sets the thermal part and explicitly sets the rest to zero. I suggest adding a comment like "Only heat is allowed to flow through this boundary" before line 624.

**Review by totto82 — APPROVED** [review]
Thanks. The code looks good. Only minor comments.

**Comment by daavid00** [comment]
jenkins build this please

**Pull request merged**